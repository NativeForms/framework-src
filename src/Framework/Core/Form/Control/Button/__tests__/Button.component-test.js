import React from 'react';
import { shallow } from 'enzyme';
import ButtonComponent from '../Button.component';
import i18n from '../Button.i18n';

const messages = i18n.en;

describe('components <ButtonComponent />', () => {
  it('renders lebel button correctly -- string type', () => {
    const props = {
      label: messages.title
    };
    const wrapper = shallow(
      <ButtonComponent {...props} />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('renders lebel button correctly -- object type', () => {
    const props = {
      label: {
        code: messages.myLabel
      }
    };
    const wrapper = shallow(
      <ButtonComponent {...props} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
