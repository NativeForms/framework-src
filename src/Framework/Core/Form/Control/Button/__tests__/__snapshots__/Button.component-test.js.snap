// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <ButtonComponent /> renders label button correctly -- object type 1`] = `
<Styled(Button)>
  <Styled(Text)>
    <FormattedMessage
      id="Button Text"
      values={Object {}}
    />
  </Styled(Text)>
</Styled(Button)>
`;

exports[`components <ButtonComponent /> renders label button correctly -- string type 1`] = `
<Styled(Button)>
  <Styled(Text)>
    Button
  </Styled(Text)>
</Styled(Button)>
`;
